### PR TITLE
Native version

### PR DESCRIPTION
**Pull request by sergeyklay**
@dreamsxin @sjinks 
Could you please take a look.

I with @ovr propose to use native zend_call_function for PHP 7.1.

As we can see for old implementation `fci.object.ce` and `executor_globals.current_execute_data.This.value.ce` is broken in `zephir_call_user_function`.
So we need get from somewhere `call` and put it to the `EG(current_execute_data)->This`

So, after long review, We can write that original `zend_function_call` fix  `EG(current_execute_data)->This`
We known that original function can fix this moment (without `fci.cache` (this can be null, but not wrong pointer :)))

1 But We cannot find moment what was changed in original function because it can fix this moment
2 We cannot find `zend_execute_internal`

### CODE REVIEW

File: ext/kernel/fcall.c [hunk 1]
```diff
@@ -632,11 +640,154 @@ int zephir_call_zval_func_aparams(zval *return_value_ptr, zval *func_name,
...
+			add_next_index_string(&func_name, "parent");
+			break;
+		case zephir_fcall_self:
+			add_next_index_string(&func_name, "self");
+			break;
+		case zephir_fcall_static:
+			add_next_index_string(&func_name, "static");
+			break;
+		case zephir_fcall_method:
+			Z_TRY_ADDREF_P(object);
```
Comment (dreamsxin) [inline ext/kernel/fcall.c]: Whether need to add to the object value judgment?

File: ext/kernel/fcall.c [hunk 2]
```diff
@@ -632,11 +640,154 @@ int zephir_call_zval_func_aparams(zval *return_value_ptr, zval *func_name,
...
+				zval_ptr_dtor(return_value_ptr);
+				ZVAL_UNDEF(return_value_ptr);
+			}
+		}
+	}
+
+	if (!return_value_ptr) {
+		zval_ptr_dtor(&rv);
+	}
+
```
Comment (dreamsxin) [inline ext/kernel/fcall.c]: add `efree(arguments)`

**Comment by dreamsxin** [comment]
I think this is a viable way.

File: ext/kernel/fcall.c (hunk 2 above)
Comment (sergeyklay) [inline ext/kernel/fcall.c]: added

**Comment by sjinks** [comment]
@sergeyklay Cannot find a better place to ask, asking here:

```c
int zephir_call_class_method_aparams(zval *return_value_ptr, zend_class_entry *ce, zephir_call_type type, zval *object,
	const char *method_name, uint method_len,
	zephir_fcall_cache_entry **cache_entry, int cache_slot,
	uint param_count, zval **params)
{
//...
	zval *fn = NULL;
        zephir_fcall_info info;
//...

#if PHP_VERSION_ID < 70100
	if (!cache_entry || !*cache_entry) {
// ...
        }
#endif

        status = zephir_call_user_function(object ? object : NULL, ce, type, fn, rvp, cache_entry, cache_slot, param_count, params, &info);
```

At this point `fn` is `NULL` if we are running under PHP 7.1 and `info` is garbage.

This seems like a bug to me, what do you think?

**Comment by sergeyklay** [comment]
Yes, looks like it is garbage.

**Comment by sergeyklay** [comment]
@sjinks Well, now here is 2 function:
```c
#if PHP_VERSION_ID >= 70100
	return zephir_call_class_method_aparams_native(
		// ...
	);
#else
	return zephir_call_class_method_aparams_fast(
		// ...
	);
#endif
```

I've partially cleaned `zephir_call_class_method_aparams_fast` and `zephir_call_class_method_aparams_native`.

So now in this proposal we use `fn` and `info` only for  PHP_VERSION_ID < 70100 in `zephir_call_class_method_aparams_fast `.

Is it what you meant, right?

**Comment by sergeyklay** [comment]
I'm closing this PR in favor of #6

**Pull request closed without merging**